nixos/modules/services/mail/postal.nix

```diff
@@ -0,0 +1,926 @@
+{
+  config,
+  lib,
+  options,
+  pkgs,
+  ...
+}:
+with lib;
+let
+  cfg = config.services.postal;
+  opt = options.services.postal;
+
+  yaml = pkgs.formats.yaml { };
+
+  postalStaticConfig = {
+    version = 2;
+    rails.environment = "production";
+  };
+
+  postalConfigFile = yaml.generate "postal-config.yml" (cfg.config // postalStaticConfig);
+
+  wrapper = pkgs.writeShellScriptBin "postal" (
+    lib.concatStrings [
+      (lib.optionalString (cfg.environmentFile != null) ''
+        export $(cat ${cfg.environmentFile} | xargs)
+      '')
+      ''
+        export POSTAL_CONFIG_FILE_PATH=${postalConfigFile}
+        exec ${pkgs.postal}/bin/postal "$@"
+      ''
+    ]
+  );
+
+  postalCli = pkgs.symlinkJoin {
+    name = "postal";
+    paths = [ wrapper ];
+  };
+
+in
+{
+  options.services.postal = {
+    enable = mkEnableOption "Enable Postal service" // {
+      description = ''
+        Enable Postal service, check [the manual](#module-services-postal)
+        for basic usage example.
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "postal";
+      description = "User to run postal and related services.";
+    };
+
+    createDatabaseLocally = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether a mariadb database should be automatically set up on the local host.
+        Set to `false` if you want to provision a database yourself.
+      '';
+    };
+
+    domain = mkOption {
+      type = types.str;
+      description = "The domain name for your postal installation.";
+      example = "postal.example.com";
+    };
+
+    environmentFile = mkOption {
+      type = with types; nullOr path;
+      default = null;
+      description = ''
+        File to load environment variables, especially secrets that cannot be set
+        via the module options. It should include `RAILS_SECRET_KEY`, which can
+        be generated with `openssl rand -hex 64`.
+        [List of available variables](https://github.com/postalserver/postal/blob/main/doc/config/environment-variables.md)
+      '';
+      example = "/run/secrets/postal.env";
+    };
+
+    workers = mkOption {
+      type = types.int;
+      default = 1;
+      description = "Number of workers to handle incoming and outgoing e-mail, to increase if you process lots of e-mails.";
+    };
+
+    # All postal config is exposed here, except secrets and postalStaticConfig attributes (declared above)
+    # https://github.com/postalserver/postal/blob/main/doc/config/yaml.yml
+    config = {
```

Comment:
Could you transform this to a `types.submodule` and rename it settings, in order to align it with [RFC42](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md). This also allows user to specify config keys without them first being added as a module option.

Reply:
Oh I didn't know about this, it's great! I removed a lot of the module options that should rarely be used, thanks!